### ISSUE TITLE
404 when creating PR from pull branch

### ISSUE THREAD

**Issue by AlexArendsen** [shayki5/azure-devops-create-pr-task#116]
Hello!

I'm attempting to build some automation that create PRs as a branch policy-- essentially, when I create a PR into `dev`, I want to make a PR for the same source branch into `test`.

When I run the pipeline with a regular, non-pull branch it works just fine, creating the PR without issue. However, when I make it create a PR from a branch like `refs/pull/4/merge`, I get:

```
##[error]The remote server returned an error: (404) Not Found.
```

I submitted a ticket to MS first, and they said I should come over here. Could I get a fix for this? Thanks much!

**Comment by shayki5** [comment]
Hi @AlexArendsen, so - you create a PR from an existing PR?
Can you run it with `system.debug` = `true` and share here the logs?

**Comment by AlexArendsen** [comment]
[create-pr-debug.txt](https://github.com/shayki5/azure-devops-create-pr-task/files/7456081/create-pr-debug.txt)

Hi @shayki5! Sorry for the delay, please see above for the logs. I had to scrub a few details for security reasons, but it should still be diagnosable.

**Comment by shayki5** [comment]
Hi @AlexArendsen, I think I found the issue, when you are in a PR, Azure DevOps create a temporary branch (`refs/pull/4/merge`), we can't merge this branch to other branches, so we can't also create a PR from this branch to other branches.
If you are in a PR you can create another PR from another source branch, but not from the branch of the PR.

**Comment by AlexArendsen** [comment]
Hi @shayki5 -- I see, I suspected that would be the issue. I fixed the problem by creating a new branch using a Bash task:

```bash
set -e
newbranch="merge-$(Build.BuildId)"
git checkout -b "${newbranch}"
git push origin "${newbranch}"
```

I had to grant our build service permission to "Create Branches" under Project Settings > Repositories > (our repo) > "Security" tab

After that, I just used your task to create a PR for `merge-$(Build.BuildId)` and it worked like a charm! Going to go ahead and close this issue-- cheers!
